js/views/userCard.js

```diff
@@ -4,19 +4,38 @@ import loadTemplate from '../utils/loadTemplate';
 import app from '../app';
 import { followedByYou, followUnfollow } from '../utils/follow';
 import Profile from '../models/profile/Profile';
+import UserCard from '../models/UserCard';
 
 export default class extends BaseVw {
   constructor(options = {}) {
     super(options);
     this.options = options;
-    this.guid = options.guid || this.model.get('guid');
+
+    if (this.model instanceof Profile) {
+      this.guid = this.model.id;
+      this.fetched = true;
+    } else if (this.model) {
+      this.guid = this.model.get('guid');
+      this.fetched = false;
+    } else {
+      this.model = new UserCard({ guid: options.guid });
```

Comment:
Isn't it a profile that we want to fetch here? The UserCard doesn't have a url function implemented, so fetching it will just be a failed call. Also, the profile model has all the functionality (parse, etc...) to handle the profile data (which I believe is what we want here).